File: source-java/guessing_game/src/main/java/xyz/zetz/fj/guessing_game/GuessingGame.java
package xyz.zetz.fj.guessing_game;

import java.io.IOException;
import java.util.Random;
import java.util.Scanner;

public class GuessingGame {
    private final int mSecretNumber = new Random().nextInt(99) + 1;

    public GuessingGame() {
    }

    public void startGaming() {
        System.out.println("Guess the number!");

        Scanner scan = new Scanner(System.in);
        int guess = 0;
        while (true) {
            System.out.println("Please input your guess.");
            if (scan.hasNextInt()) {
                guess = scan.nextInt();
            }

            System.out.println("Your guessed: " + guess);

            if (guess == mSecretNumber) {
                System.out.println("You win!");
                break;
            } else if (guess < mSecretNumber) {
                System.out.println("Too small!");
            } else {
                System.out.println("Too big!");
            }
        }
        scan.close();
    }
}

File: source-lang/c/guessing_game/guessing_game.c
#include "zetz/guessing_game.h"

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

static int s_secret_number = 0;

void zetz_guessing_game_init()
{
    srand((unsigned)time(NULL));
    s_secret_number = rand() % 100;
}

void zetz_guessing_game_run()
{
    printf("Guess the number!\n");

    int guess = 0;
    while (1)
    {
        printf("Please input your guess.\n");
        scanf("%d", &guess);

        printf("You guessed: %d\n", guess);

        if (guess == s_secret_number)
        {
            printf("You win!\n");
            break;
        }
        else if (guess < s_secret_number)
        {
            printf("Too small!\n");
        }
        else
        {
            printf("Too big!\n");
        }
    }
}